### ISSUE TITLE
Alexell's splash image

### ISSUE THREAD

**Issue by SevenSpheres** [CelestiaProject/Celestia#438]
In 2016 Alexell made a new Celestia splash image (last one in [this](https://celestia.space/forum/viewtopic.php?p=135210#p135210) post). Should it be used in version 1.7?

**Comment by 375gnu** [comment]
Qt has longstanding issues with transparent images. It doesn't support 8bit transparency, so in some cases we have solid black colour instead of transparency. If somebody fixes our Qt splashscreen, we'll consider including one of those images.

**Comment by SevenSpheres** [comment]
The current splash image also has transparency...are you saying it will need to change?

**Comment by 375gnu** [comment]
@SevenSpheres yes, but the current image has quite small transparent border, while Alexell's one has large regions. So with the current one this issue is not much noticeable.

**Comment by SevenSpheres** [comment]
I don't see much of a difference in the size of the transparent regions. Doesn't really matter what splash image is used of course.

**Comment by 375gnu** [comment]
I'll check it, I have that issue on my computer.

**Comment by 375gnu** [comment]
We can vote on the forum:) And check Alexell's license for those images.

**Comment by Alexell** [comment]
@375gnu if necessary, you can use these images. They were made specifically for Celestia.

**Comment by 375gnu** [comment]
@Alexell we need a clear license. I propose the same license we use for the code - GPLv2 or later

**Comment by Alexell** [comment]
@375gnu ok i agree

**Comment by SevenSpheres** [comment]
Actually, why not add a line like this to celestia.cfg:
`  SplashImage  "splash.png"`
Then the user can choose what splash image they want without having to rename/replace any files.
Celestia could even install with multiple images for the user to choose from.

It would also be nice to be able to put other windows in front of the splash screen.

**Comment by 375gnu** [comment]
On 10/12/19, SevenSpheres <notifications@github.com> wrote:
> Actually, why not add a line like this to celestia.cfg:
> `  SplashImage  "splash.png"`

Good idea.

> It would also be nice to be able to put other windows in front of the splash
> screen.

Why? The main idea of splash screens is that a user sees that an
application is working while it's not ready to interact with a user.

**Comment by SevenSpheres** [comment]
> Why? The main idea of splash screens is that a user sees that an
> application is working while it's not ready to interact with a user.

Because if you're using Celestia with lots of addons, it takes a long time to load, and since the splash screen is always in front you can't do anything in other windows.

**Comment by 375gnu** [comment]
but it's not ready yet, what will you do?

**Comment by SevenSpheres** [comment]
[I asked about this on the forum.](https://celestia.space/forum/viewtopic.php?f=4&t=19982)

**Comment by SevenSpheres** [comment]
Renamed this issue because it's been derailed from Alexell's image...

Anyway 6 people (including **pirogronian**) agree with me that the splash screen shouldn't always be in front. **FarGetaNik** also notes that the latest 1.7 doesn't display the splash screen at all, but I assume this will be fixed?

**Comment by pirogronian** [comment]
On my machine (Linux) there is no problem: splash screen can be hidden by other windows.

**Comment by SevenSpheres** [comment]
I guess I'll wait until the splash screen is restored in the 1.7 Windows version and then see if this has been fixed.

**Comment by SevenSpheres** [comment]
I'm not really sure what to do with this thread. Here's a checklist of things I've suggested doing; should I split them into other issues instead?
- [ ] Make it possible to override splash in Windows version
- [ ] Restore splash in QT version
- [ ] Add SplashImage line to celestia.cfg
- [ ] Add more splash images (e.g., Alexell's)

**Comment by 375gnu** [comment]
in Qt version splash is searched in c:/Program Files (x86)/celestia/share/celestia/splash/splash.png, i.e. where celestia should be installed

**Comment by 375gnu** [comment]
let's keep your suggestions here and reference it from pull requests

**Comment by SevenSpheres** [comment]
> in Qt version splash is searched in c:/Program Files (x86)/celestia/share/celestia/splash/splash.png, i.e. where celestia should be installed

In the QT version the splash screen is not displayed. Also, on Windows the above directory structure does not exist; splash.png is in the main Celestia directory. Should it be in share/celestia/splash instead?

**Comment by 375gnu** [comment]
> Should it be in share/celestia/splash instead?

yes

**Comment by SevenSpheres** [comment]
I created a directory share/celestia/splash and moved splash.png there. There is no change in the 1.7 QT version (i.e., still no splash screen). In the 1.7 Win version the splash screen is gone.

**Comment by 375gnu** [comment]
ok, we'll need to handle it in both frontends

**Comment by SevenSpheres** [comment]
So in the latest build (Win version) the splash screen can be overridden, but it doesn't work right. When another window is brought in front of the splash, instead of the splash going behind just that one window, it goes all the way to the back, i.e. behind all windows.

**Comment by 375gnu** [comment]
None of us uses windows. See #322 . Don't expect native win frontend to work 100% correctly.

**Comment by SevenSpheres** [comment]
The splash image appears to have been restored in the Qt version! 👍

**Comment by 375gnu** [comment]
regarding the `splashimage`option. the problem is that some interfaces (qt, win32) create splash image before they start loading different data, only gtk first creates `CelestiaCore` which loads `celestia.cfg` before creating a splash. personally i'm not eager not rewrite qt ui (it *looks* like fixing win32 is easy).
